Add doc updates (#3347)
kubernetes-sigs/aws-load-balancer-controller: docs/guide/ingress/spec.md

# Ingress specification
This document covers how ingress resources work in relation to The AWS Load Balancer Controller.

!!!note ""

>     - Beginning from v2.4.3 of the AWS LBC, rules of `pathType: ImplementationSpecific` are always ordered according to the rule order in the manifest.
>     - However, in case of mixed path types,  

Reviewer: The second part of this sentence is redundant with the second list item:

```suggestion
    Beginning from v2.4.3 of the AWS LBC, rules are ordered as follows:
```

>         - `pathType: Exact` match are always ordered first 

Reviewer: ```suggestion
        - `pathType: Exact` paths are always ordered first 
```

>         - followed by `pathType: Prefix` paths with the longest prefix first

Reviewer: ```suggestion
        - followed by `pathType: Prefix` paths, with the longest prefix first
```

>         - `pathType: ImplementationSpecific` paths in the order from the manifest

Reviewer: ```suggestion
        - followed by `pathType: ImplementationSpecific` paths, in the order they are listed in the manifest
```
